### PR TITLE
Ignore default security context constraints when running on kube

### PR DESCRIPTION
**Pull request by derekwaynecarr**
If you are running OpenShift on a Kubernetes cluster, we need to ignore security context constraints since they are not actually in upstream Kubernetes...

### CODE REVIEW

**Comment by deads2k** [comment]
Is it too painful to make this decision based on discovery information from the kube server?  That seems more correct than basing on whether or not we're running an embedded kube.

**Comment by liggitt** [comment]
this really just ignores errors we tolerate... is it worth it?

**Comment by paralin** [comment]
I noticed these errors when running openshift on an existing kubernetes server. So +1 for merging something similar to this.

**Comment by liggitt** [comment]
To be clear, it is only logging, it shouldn't affect anything functionally

**Comment by deads2k** [comment]
> this really just ignores errors we tolerate... is it worth it?

Depends on the pain.  I don't feel strongly for a one-off, but I worry about this becoming something like the atomic switch we had that rippled through for builds.

**Comment by derekwaynecarr** [comment]
* Fixed the gofmt issue (embarrasing that VS Code was not gofmt-ing origin yet).
* Added a TODO to move to discovery client.

As for the atomic switch comparison, thus far this is the only change I have identified where running OpenShift on Kubernetes can cause confusing log output.  Of course, some features just don't work, but for the experimental command we are trying to put together, I think it's fine.

**Comment by derekwaynecarr** [comment]
eh, thinking about it should not be too hard to switch to discovery client.  will try that now.

**Comment by derekwaynecarr** [comment]
@deads2k @liggitt - now with discovery!  PTAL

File: pkg/cmd/server/origin/ensure.go [hunk 1]
```diff
@@ -151,7 +153,47 @@ func (c *MasterConfig) ensureNamespaceServiceAccountRoleBindings(namespace *kapi
 	}
 }
 
+func (c *MasterConfig) securityContextConstraintsSupported() (bool, error) {
```
Comment (deads2k) [inline pkg/cmd/server/origin/ensure.go]: Add TODO to make this a library upstream, `ResourceExists(GroupVersionResource)` or some such.  Definitely not a blocker, just seen a few these ripple through.

**Comment by deads2k** [comment]
lgtm.

**Comment by derekwaynecarr** [comment]
Any objections to merging?

**Comment by paralin** [comment]
Seems good to me, I merged this into my working tree a few days ago, no problems

**Comment by deads2k** [comment]
[merge]

**Comment by bparees** [comment]
[merge] vagrant plugin was broken.

**Pull request merged**